src/main/java/BMO/Bmo.java

```diff
@@ -0,0 +1,146 @@
+package bmo;
+
+import java.util.Scanner;
+
+import bmo.task.Task;
+
+/**
+ * Bmo is a chatbot that helps users manage their tasks.
+ */
+public class Bmo {
+
+    private static String filePath = "ip/data/BMO.txt";
+    private Storage storage;
+    private Ui ui;
+    private Parser parser;
+    private TaskList taskList;
+
+    /**
+     * Constructor for Bmo class.
+     *
+     * @param filePath the path of the storage file
+     * @throws Exception if unable to create the storage file
+     */
+    public Bmo(String filePath) throws Exception {
+        try {
+            this.storage = new Storage(filePath);
+            this.ui = new Ui();
+            this.parser = new Parser();
+            this.taskList = new TaskList();
+            Bmo.filePath = filePath;
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    /**
+     * Runs the BMO chatbot program
+     *
+     * @throws IOException if unable to read or write to file
+     * @throws BmoException if the user input is invalid
+     */
+    public void run() throws Exception {
+        storage.readStorageFile(taskList, filePath);
+        ui.printWelcome();
+        Scanner sc = new Scanner(System.in);
+
+        loop:
+        while (true) {
```

Comment:
A suggestion that I can give is you can create a ProcessTask class that can handle each type of process like adding, marking, deleting, etc. Makes it easy for testing later.